examples/sw/simple_system/common/simple_system_common.c

```diff
@@ -139,3 +139,37 @@ void simple_exc_handler(void) {
 
   while(1);
 }
+
+volatile uint64_t time_elapsed;
+uint32_t time_increment;
+
+inline static void increment_timecmp(uint32_t time_base) {
+  uint64_t current_time = (uint64_t)DEV_READ(TIMER_BASE + TIMER_MTIMEH, 0);
+  current_time <<= 32;
+  current_time |= (uint64_t)DEV_READ(TIMER_BASE + TIMER_MTIME, 0);
+  current_time += (uint64_t)time_base;
+  DEV_WRITE(TIMER_BASE + TIMER_MTIMECMPH, current_time >> 32);
+  DEV_WRITE(TIMER_BASE + TIMER_MTIMECMP, current_time);
```

Comment:
I was originally trying to keep the handler as small as possible and so wanted to avoid the extra instructions (the function is only used while an interrupt is already pending or is disabled so no spurious interrupt), but you're right it makes sense to do what the spec says, and it's only two extra instructions!

Reply:
Thanks for changing this.